hedera-node/hedera-app/src/test/java/com/hedera/node/app/authorization/PrivilegesVerifierTest.java

```diff
@@ -0,0 +1,678 @@
+// (It was a good thing. I discovered two bugs...) :)
+class PrivilegesVerifierTest {
+
+    private Wrapper subject;
+
+    private static class Wrapper {
+        private final PrivilegesVerifier delegate;
+
+        Wrapper(final ConfigProvider configProvider) {
+            delegate = new PrivilegesVerifier(configProvider);
+        }
+
+        SystemOpAuthorization checkAccessor(final SignedTxnAccessor accessor) {
+            com.hedera.hapi.node.base.AccountID accountID = PbjConverter.toPbj(accessor.getPayer());
+            com.hedera.hapi.node.base.HederaFunctionality functionality = PbjConverter.toPbj(accessor.getFunction());
+            com.hedera.hapi.node.transaction.TransactionBody txBody = PbjConverter.toPbj(accessor.getTxn());
+            final var pbjResult = delegate.hasPrivileges(accountID, functionality, txBody);
+            return SystemOpAuthorization.valueOf(pbjResult.name());
+        }
+
+        boolean canPerformNonCryptoUpdate(final long accountNum, final long fileNum) {
+            final var accountID = com.hedera.hapi.node.base.AccountID.newBuilder()
+                    .accountNum(accountNum)
+                    .build();
+            final var fileID = com.hedera.hapi.node.base.FileID.newBuilder()
+                    .fileNum(fileNum)
+                    .build();
+            final var fileUpdateTxBody = com.hedera.hapi.node.transaction.TransactionBody.newBuilder()
+                    .fileUpdate(com.hedera.hapi.node.file.FileUpdateTransactionBody.newBuilder()
+                            .fileID(fileID)
+                            .build())
+                    .build();
+            final var fileAppendTxBody = com.hedera.hapi.node.transaction.TransactionBody.newBuilder()
+                    .fileAppend(com.hedera.hapi.node.file.FileAppendTransactionBody.newBuilder()
+                            .fileID(fileID)
+                            .build())
+                    .build();
+            return delegate.hasPrivileges(accountID, HederaFunctionality.FILE_UPDATE, fileUpdateTxBody)
+                            == SystemPrivilege.AUTHORIZED
+                    && delegate.hasPrivileges(accountID, HederaFunctionality.FILE_APPEND, fileAppendTxBody)
+                            == SystemPrivilege.AUTHORIZED;
+        }
+    }
+
+    @BeforeEach
+    void setUp() {
+        final var configuration = HederaTestConfigBuilder.createConfig();
+        final ConfigProvider configProvider = () -> new VersionedConfigImpl(configuration, 1L);
+
+        subject = new Wrapper(configProvider);
+    }
+
+    @SuppressWarnings("DataFlowIssue")
+    @Test
+    void testConstructorWithIllegalParameters() {
+        assertThatThrownBy(() -> new PrivilegesVerifier(null)).isInstanceOf(NullPointerException.class);
+    }
+
+    @Test
+    void treasuryCanUpdateAllNonAccountEntities() {
+        // expect:
+        assertTrue(subject.canPerformNonCryptoUpdate(2, 101));
```

Comment:
These would be much clearer if the numbers were constants named for the "file" being tested (e.g. `ADDRESS_BOOK_NUM` or `THROTTLES_NUM`).